From ce62803773cec394d738f64379d9b6bf816f497f Mon Sep 17 00:00:00 2001
From: Conor Dooley <conor.dooley@microchip.com>
Date: Tue, 2 Nov 2021 15:19:19 +0000
Subject: [PATCH 70/93] riscv: dts: mpfs, icicle kit: minor updates and fixes

update mmc/sd speed options & drop some duplicates,
enable i2c0, add missing interrupt at 117 & update
f2h defines where relevant.

mpfs-device-tree commit-id: 5b2ed676915538f52aaee46dd2e90a9b9c33df43
---
 .../dts/microchip/microchip-mpfs-fabric.dtsi  |   7 +-
 .../microchip-mpfs-icicle-kit-context-a.dts   |  15 ++
 .../microchip/microchip-mpfs-icicle-kit.dts   |  15 ++
 .../boot/dts/microchip/microchip-mpfs.dtsi    |   7 +-
 .../microchip,mpfs-plic.h                     | 140 +++++++++---------
 5 files changed, 106 insertions(+), 78 deletions(-)

diff --git a/arch/riscv/boot/dts/microchip/microchip-mpfs-fabric.dtsi b/arch/riscv/boot/dts/microchip/microchip-mpfs-fabric.dtsi
index 463358fdebac..4b370690292b 100644
--- a/arch/riscv/boot/dts/microchip/microchip-mpfs-fabric.dtsi
+++ b/arch/riscv/boot/dts/microchip/microchip-mpfs-fabric.dtsi
@@ -8,13 +8,14 @@ fpgadma: fpgadma@60020000 {
 		#size-cells = <0>;
 		reg = <0x0 0x60020000 0x0 0x1000>;
 		interrupt-parent = <&plic>;
-		interrupts = <PLIC_INT_FABRIC_F2H_3>;
+		interrupts = <PLIC_INT_FABRIC_F2H_2>;
 		status = "okay";
 	};
 
-	fpgalsram: fpgalsram@61000000 {
+	fpgalsram: fpga_lsram@61000000 {
 		compatible = "generic-uio";
-		reg = < 0x0 0x61000000 0x0 0x10000>;
+		reg = <0x0 0x61000000 0x0 0x0001000
+			0x14 0x00000000 0x0 0x00010000>;
 		status = "okay";
 	};
 };
diff --git a/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit-context-a.dts b/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit-context-a.dts
index dbbb807b3569..04311dca997d 100644
--- a/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit-context-a.dts
+++ b/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit-context-a.dts
@@ -112,6 +112,17 @@ &mmuart4 {
 
 &mmc {
 	status = "okay";
+	bus-width = <4>;
+	disable-wp;
+	cap-sd-highspeed;
+	cap-mmc-highspeed;
+	card-detect-delay = <200>;
+	mmc-ddr-1_8v;
+	mmc-hs200-1_8v;
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
 };
 
 &spi0 {
@@ -132,6 +143,10 @@ &qspi {
 	status = "okay";
 };
 
+&i2c0 {
+	status = "okay";
+};
+
 &i2c1 {
 	status = "okay";
 	pac193x: pac193x@10 {
diff --git a/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit.dts b/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit.dts
index c6e81aeefb1c..5e62533575fc 100644
--- a/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit.dts
+++ b/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit.dts
@@ -112,6 +112,17 @@ &mmuart4 {
 
 &mmc {
 	status = "okay";
+	bus-width = <4>;
+	disable-wp;
+	cap-sd-highspeed;
+	cap-mmc-highspeed;
+	card-detect-delay = <200>;
+	mmc-ddr-1_8v;
+	mmc-hs200-1_8v;
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
 };
 
 &spi0 {
@@ -132,6 +143,10 @@ &qspi {
 	status = "okay";
 };
 
+&i2c0 {
+	status = "okay";
+};
+
 &i2c1 {
 	status = "okay";
 	pac193x: pac193x@10 {
diff --git a/arch/riscv/boot/dts/microchip/microchip-mpfs.dtsi b/arch/riscv/boot/dts/microchip/microchip-mpfs.dtsi
index cf49c8f7c9b4..34f1ef49beb9 100644
--- a/arch/riscv/boot/dts/microchip/microchip-mpfs.dtsi
+++ b/arch/riscv/boot/dts/microchip/microchip-mpfs.dtsi
@@ -290,11 +290,6 @@ mmc: mmc@20008000 { /* Common node entry for emmc/sd */
 			clocks = <&clkcfg CLK_MMC>;
 			interrupt-parent = <&plic>;
 			interrupts = <PLIC_INT_MMC_MAIN PLIC_INT_MMC_WAKEUP>;
-			bus-width = <4>;
-			cap-mmc-highspeed;
-			cap-sd-highspeed;
-			no-1-8-v;
-			disable-wp;
 			max-frequency = <200000000>;
 			status = "disabled";
 		};
@@ -498,7 +493,7 @@ pcie: pcie@2000000000 {
 			clock-names = "fic0", "fic1", "fic3";
 			bus-range = <0x0 0x7f>;
 			interrupt-parent = <&plic>;
-			interrupts = <PLIC_INT_FABRIC_F2H_2>;
+			interrupts = <PLIC_INT_FABRIC_F2H_1>;
 			interrupt-map = <0 0 0 1 &pcie_intc 0>,
 					<0 0 0 2 &pcie_intc 1>,
 					<0 0 0 3 &pcie_intc 2>,
diff --git a/include/dt-bindings/interrupt-controller/microchip,mpfs-plic.h b/include/dt-bindings/interrupt-controller/microchip,mpfs-plic.h
index ff6b3ab25e74..81c8cd02abd8 100755
--- a/include/dt-bindings/interrupt-controller/microchip,mpfs-plic.h
+++ b/include/dt-bindings/interrupt-controller/microchip,mpfs-plic.h
@@ -124,74 +124,76 @@
 #define PLIC_INT_ATHENA_BUS_ERROR				114
 #define PLIC_INT_USOC_AXIC_US					115
 #define PLIC_INT_USOC_AXIC_DS					116
-#define PLIC_INT_FABRIC_F2H_0					117
-#define PLIC_INT_FABRIC_F2H_1					118
-#define PLIC_INT_FABRIC_F2H_2					119
-#define PLIC_INT_FABRIC_F2H_3					120
-#define PLIC_INT_FABRIC_F2H_4					121
-#define PLIC_INT_FABRIC_F2H_5					122
-#define PLIC_INT_FABRIC_F2H_6					123
-#define PLIC_INT_FABRIC_F2H_7					124
-#define PLIC_INT_FABRIC_F2H_8					125
-#define PLIC_INT_FABRIC_F2H_9					126
-#define PLIC_INT_FABRIC_F2H_10					127
-#define PLIC_INT_FABRIC_F2H_11					128
-#define PLIC_INT_FABRIC_F2H_12					129
-#define PLIC_INT_FABRIC_F2H_13					130
-#define PLIC_INT_FABRIC_F2H_14					131
-#define PLIC_INT_FABRIC_F2H_15					132
-#define PLIC_INT_FABRIC_F2H_16					133
-#define PLIC_INT_FABRIC_F2H_17					134
-#define PLIC_INT_FABRIC_F2H_18					135
-#define PLIC_INT_FABRIC_F2H_19					136
-#define PLIC_INT_FABRIC_F2H_20					137
-#define PLIC_INT_FABRIC_F2H_21					138
-#define PLIC_INT_FABRIC_F2H_22					139
-#define PLIC_INT_FABRIC_F2H_23					140
-#define PLIC_INT_FABRIC_F2H_24					141
-#define PLIC_INT_FABRIC_F2H_25					142
-#define PLIC_INT_FABRIC_F2H_26					143
-#define PLIC_INT_FABRIC_F2H_27					144
-#define PLIC_INT_FABRIC_F2H_28					145
-#define PLIC_INT_FABRIC_F2H_29					146
-#define PLIC_INT_FABRIC_F2H_30					147
-#define PLIC_INT_FABRIC_F2H_31					148
-#define PLIC_INT_FABRIC_F2H_32					149
-#define PLIC_INT_FABRIC_F2H_33					150
-#define PLIC_INT_FABRIC_F2H_34					151
-#define PLIC_INT_FABRIC_F2H_35					152
-#define PLIC_INT_FABRIC_F2H_36					153
-#define PLIC_INT_FABRIC_F2H_37					154
-#define PLIC_INT_FABRIC_F2H_38					155
-#define PLIC_INT_FABRIC_F2H_39					156
-#define PLIC_INT_FABRIC_F2H_40					157
-#define PLIC_INT_FABRIC_F2H_41					158
-#define PLIC_INT_FABRIC_F2H_42					159
-#define PLIC_INT_FABRIC_F2H_43					160
-#define PLIC_INT_FABRIC_F2H_44					161
-#define PLIC_INT_FABRIC_F2H_45					162
-#define PLIC_INT_FABRIC_F2H_46					163
-#define PLIC_INT_FABRIC_F2H_47					164
-#define PLIC_INT_FABRIC_F2H_48					165
-#define PLIC_INT_FABRIC_F2H_49					166
-#define PLIC_INT_FABRIC_F2H_50					167
-#define PLIC_INT_FABRIC_F2H_51					168
-#define PLIC_INT_FABRIC_F2H_52					169
-#define PLIC_INT_FABRIC_F2H_53					170
-#define PLIC_INT_FABRIC_F2H_54					171
-#define PLIC_INT_FABRIC_F2H_55					172
-#define PLIC_INT_FABRIC_F2H_56					173
-#define PLIC_INT_FABRIC_F2H_57					174
-#define PLIC_INT_FABRIC_F2H_58					175
-#define PLIC_INT_FABRIC_F2H_59					176
-#define PLIC_INT_FABRIC_F2H_60					177
-#define PLIC_INT_FABRIC_F2H_61					178
-#define PLIC_INT_FABRIC_F2H_62					179
-#define PLIC_INT_FABRIC_F2H_63					180
-#define PLIC_INT_BUS_ERROR_UNIT_HART_0				181
-#define PLIC_INT_BUS_ERROR_UNIT_HART_1				182
-#define PLIC_INT_BUS_ERROR_UNIT_HART_2				183
-#define PLIC_INT_BUS_ERROR_UNIT_HART_3				184
-#define PLIC_INT_BUS_ERROR_UNIT_HART_4				185
+#define PLIC_INT_SPARE						117
+
+#define PLIC_INT_FABRIC_F2H_0					118
+#define PLIC_INT_FABRIC_F2H_1					119
+#define PLIC_INT_FABRIC_F2H_2					120
+#define PLIC_INT_FABRIC_F2H_3					121
+#define PLIC_INT_FABRIC_F2H_4					122
+#define PLIC_INT_FABRIC_F2H_5					123
+#define PLIC_INT_FABRIC_F2H_6					124
+#define PLIC_INT_FABRIC_F2H_7					125
+#define PLIC_INT_FABRIC_F2H_8					126
+#define PLIC_INT_FABRIC_F2H_9					127
+#define PLIC_INT_FABRIC_F2H_10					128
+#define PLIC_INT_FABRIC_F2H_11					129
+#define PLIC_INT_FABRIC_F2H_12					130
+#define PLIC_INT_FABRIC_F2H_13					131
+#define PLIC_INT_FABRIC_F2H_14					132
+#define PLIC_INT_FABRIC_F2H_15					133
+#define PLIC_INT_FABRIC_F2H_16					134
+#define PLIC_INT_FABRIC_F2H_17					135
+#define PLIC_INT_FABRIC_F2H_18					136
+#define PLIC_INT_FABRIC_F2H_19					137
+#define PLIC_INT_FABRIC_F2H_20					138
+#define PLIC_INT_FABRIC_F2H_21					139
+#define PLIC_INT_FABRIC_F2H_22					140
+#define PLIC_INT_FABRIC_F2H_23					141
+#define PLIC_INT_FABRIC_F2H_24					142
+#define PLIC_INT_FABRIC_F2H_25					143
+#define PLIC_INT_FABRIC_F2H_26					144
+#define PLIC_INT_FABRIC_F2H_27					145
+#define PLIC_INT_FABRIC_F2H_28					146
+#define PLIC_INT_FABRIC_F2H_29					147
+#define PLIC_INT_FABRIC_F2H_30					148
+#define PLIC_INT_FABRIC_F2H_31					149
+#define PLIC_INT_FABRIC_F2H_32					150
+#define PLIC_INT_FABRIC_F2H_33					151
+#define PLIC_INT_FABRIC_F2H_34					152
+#define PLIC_INT_FABRIC_F2H_35					153
+#define PLIC_INT_FABRIC_F2H_36					154
+#define PLIC_INT_FABRIC_F2H_37					155
+#define PLIC_INT_FABRIC_F2H_38					156
+#define PLIC_INT_FABRIC_F2H_39					157
+#define PLIC_INT_FABRIC_F2H_40					158
+#define PLIC_INT_FABRIC_F2H_41					159
+#define PLIC_INT_FABRIC_F2H_42					160
+#define PLIC_INT_FABRIC_F2H_43					161
+#define PLIC_INT_FABRIC_F2H_44					162
+#define PLIC_INT_FABRIC_F2H_45					163
+#define PLIC_INT_FABRIC_F2H_46					164
+#define PLIC_INT_FABRIC_F2H_47					165
+#define PLIC_INT_FABRIC_F2H_48					166
+#define PLIC_INT_FABRIC_F2H_49					167
+#define PLIC_INT_FABRIC_F2H_50					168
+#define PLIC_INT_FABRIC_F2H_51					169
+#define PLIC_INT_FABRIC_F2H_52					170
+#define PLIC_INT_FABRIC_F2H_53					171
+#define PLIC_INT_FABRIC_F2H_54					172
+#define PLIC_INT_FABRIC_F2H_55					173
+#define PLIC_INT_FABRIC_F2H_56					174
+#define PLIC_INT_FABRIC_F2H_57					175
+#define PLIC_INT_FABRIC_F2H_58					176
+#define PLIC_INT_FABRIC_F2H_59					177
+#define PLIC_INT_FABRIC_F2H_60					178
+#define PLIC_INT_FABRIC_F2H_61					179
+#define PLIC_INT_FABRIC_F2H_62					180
+#define PLIC_INT_FABRIC_F2H_63					181
+#define PLIC_INT_BUS_ERROR_UNIT_HART_0				182
+#define PLIC_INT_BUS_ERROR_UNIT_HART_1				183
+#define PLIC_INT_BUS_ERROR_UNIT_HART_2				184
+#define PLIC_INT_BUS_ERROR_UNIT_HART_3				185
+#define PLIC_INT_BUS_ERROR_UNIT_HART_4				186
 
 #endif /* _DT_BINDINGS_INTERRUPT_CONTROLLER_MICROCHIP_MPFS_PLIC_H */
-- 
2.30.2

